config/configtls/clientcasfilereloader.go

```diff
@@ -112,7 +112,7 @@ func (r *clientCAsFileReloader) handleWatcherEvents() {
 			// NOTE: k8s configmaps uses symlinks, we need this workaround.
 			// original configmap file is removed.
 			// SEE: https://martensson.io/go-fsnotify-and-kubernetes-configmaps/
-			if event.Op == fsnotify.Remove || event.Op == fsnotify.Chmod {
+			if event.Has(fsnotify.Remove) || event.Has(fsnotify.Chmod) {
```

Comment:
My confusion here is why we need to monitor for the chmod...

Reply:
Possibly because of this https://github.com/fsnotify/fsnotify/issues/421?

Reply:
Inotify generates `IN_DELETE_SELF` for the watched file only when its link count is reduces to 0.  Looks like it is used to handle this situation. :thinking:
https://github.com/fsnotify/fsnotify/blob/v1.6.0/backend_inotify.go#L26-L33